elfi/distributions.py

```diff
@@ -32,6 +32,10 @@ def pdf(self, x, *params, **kwargs):
     def logpdf(self, x, *params, **kwargs):
         raise NotImplementedError
 
+    @property
+    def name(self):
+        return self._name or self.__class__.__name__
+
 
```

Comment:
I think it's simpler to instruct users to derive from elfi.Distribution. Then we can also potentially add stuff there in the future.